File: Java/DataStructures/MyQueue.java
package Java.DataStructures;

public class MyQueue {
    
    private int queue[];
    private int capacity;
    private int front;
    private int rear;

    public MyQueue(int size){
        queue = new int[size];
        capacity = size;
        front = -1;
        rear = -1;
    }

    public void enQueue(int item){

        if(isFull()){
            throw new RuntimeException("Queue is full");
        }

        if(front == -1) front++;

        rear++;
        queue[rear] = item;

    }

    public int deQueue(){

        if(isEmpty()){
            throw new RuntimeException("Queue is empty");
        }

        int frontElement = queue[front];
        front++;

        return frontElement;
    }

    public Boolean isEmpty(){
        return front == rear;
    }

    public Boolean isFull(){
        return rear == capacity - 1;
    }

    public int peek(){
        
        if(isEmpty()){
            throw new RuntimeException("Queue is empty");
        }

        return queue[front];
    }

    public void printQueue(){
        String queueAsString = "[";

        if(!isEmpty()){
            for(int i = front; i <= rear; i++){
                if(i == front){
                    queueAsString = queueAsString  + Integer.toString(queue[i]);
                }else{
                    queueAsString = queueAsString + " | " + Integer.toString(queue[i]);
                }
            }
        }

        queueAsString = queueAsString + "] <- Rear";
        System.out.println(queueAsString);
    }

}

File: Python/Data Structures/MyQueue.py
class MyQueue:
    
    def __init__(self, size):
        self.capacity = size
        self.front = -1
        self.rear = -1
        self.queue = [None] * size
    
    def enQueue(self, item):
        
        if self.isFull():
            raise Exception("Queue is Full.")

        if self.front == -1:
            self.front = self.front + 1
        
        self.rear = self.rear + 1

        self.queue[self.rear] = item

    def deQueue(self):
        
        if self.isEmpty():
            raise Exception("Queue is Empty.")
        
        frontElement = self.queue[self.front]
        self.front = self.front + 1

        return frontElement

    def isEmpty(self):
        return self.rear == self.front

    def isFull(self):
        return self.rear == self.capacity - 1

    def peek(self):
        
        if self.isEmpty():
            raise Exception("Queue is Empty.")
        
        return self.queue[self.front]

    def __str__(self):
        
        if self.isEmpty():
            return '[]'
        
        return f'{[self.queue[x] for x in range(self.front, self.rear + 1)]} <- Rear'
